modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -459,6 +459,11 @@ private void removeRemoteSubscriptionFromMemory(AndesSubscription subscription)
 
         Set<AndesSubscription> filteredSubscriptions = new HashSet<>();
         String messageRouter = destinationType.getAndesMessageRouter();
+
+        if (ProtocolType.MQTT.equals(protocolType)) {
```

Comment:
This is done because we do not have a mqtt_topic / jms_topic separation when querying at the database level. 

Introducing such a separation at the DestinationType Enum would incur a lot of mqtt-specific code, and given that we are not supporting mqtt in the future releases, we are resorting to this approach.